File: android-native/app/src/main/java/flexr/social/app/core/designsystem/icon/FlexrIcons.kt
package flexr.social.app.core.designsystem.icon

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.automirrored.filled.Send
import androidx.compose.material.icons.filled.Block
import androidx.compose.material.icons.filled.CameraAlt
import androidx.compose.material.icons.filled.Chat
import androidx.compose.material.icons.filled.EmojiEmotions
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.HeartBroken
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.MoreVert
import androidx.compose.material.icons.filled.OutlinedFlag
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.Place
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.SolidColor
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.graphics.StrokeJoin
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.graphics.vector.path
import androidx.compose.ui.unit.dp

/**
 * Marken-Symbol: die FLEXR-Hantel — dieselbe Geometrie wie das Launcher-Icon,
 * abgeleitet aus dem im Play Store hinterlegten Icon (zwei schmale äußere
 * Scheiben, zwei breite innere, durchgehender Steg). Als [ImageVector] ist es
 * überall scharf und lässt sich wie jedes Material-Icon einfärben.
 *
 * In der Oberfläche etwas leichter gestrichen als auf dem Launcher-Icon, damit
 * es bei 22dp in der Navigation nicht zuläuft.
 */
val FlexrDumbbell: ImageVector by lazy {
    ImageVector.Builder(
        name = "FlexrDumbbell",
        defaultWidth = 24.dp,
        defaultHeight = 24.dp,
        viewportWidth = 24f,
        viewportHeight = 24f,
    ).apply {
        val stroke = SolidColor(Color.White)
        fun bar(x: Float, top: Float, bottom: Float, width: Float) {
            path(
                stroke = stroke,
                strokeLineWidth = width,
                strokeLineCap = StrokeCap.Round,
                strokeLineJoin = StrokeJoin.Round,
            ) {
                moveTo(x, top)
                lineTo(x, bottom)
            }
        }
        bar(4.2f, 7.2f, 16.8f, 1.9f)    // äußere Scheibe links
        bar(8.2f, 4.6f, 19.4f, 2.6f)    // innere Scheibe links
        bar(15.8f, 4.6f, 19.4f, 2.6f)   // innere Scheibe rechts
        bar(19.8f, 7.2f, 16.8f, 1.9f)   // äußere Scheibe rechts
        path(
            stroke = stroke,
            strokeLineWidth = 2f,
            strokeLineCap = StrokeCap.Round,
        ) {
            moveTo(3.4f, 12f)
            lineTo(20.6f, 12f)
        }
    }.build()
}

/**
 * Symbolsatz der App an einer Stelle gebündelt. Wo Material Design bereits ein
 * passendes, den Nutzern vertrautes Symbol liefert, wird es verwendet — das ist
 * auf Android die richtige Wahl gegenüber nachgebauten Web-Icons.
 */
object FlexrIcons {
    val Swipe = FlexrDumbbell
    val Gym = FlexrDumbbell
    val Matches = Icons.Filled.Favorite
    val Chats = Icons.Filled.Chat
    val Account = Icons.Filled.Person
    val Like = Icons.Filled.Favorite
    val Pass = Icons.Filled.Close
    val Unmatch = Icons.Filled.HeartBroken
    val Report = Icons.Filled.OutlinedFlag
    val Block = Icons.Filled.Block
    val Back = Icons.AutoMirrored.Filled.ArrowBack
    val Send = Icons.AutoMirrored.Filled.Send
    val More = Icons.Filled.MoreVert
    val Camera = Icons.Filled.CameraAlt
    val Locked = Icons.Filled.Lock
    val Place = Icons.Filled.Place
    val Close = Icons.Filled.Close
    val Emoji = Icons.Filled.EmojiEmotions
}

/** Merkt sich das Hantel-Icon über Rekompositionen hinweg. */
@Composable
fun rememberDumbbellIcon(): ImageVector = remember { FlexrDumbbell }

File: ios/FLEXR/Core/DesignSystem/Icon/FlexrIcons.swift
import SwiftUI

/// Marken-Symbol: die FLEXR-Hantel — dieselbe Geometrie wie in der Android-App
/// (zwei schmale äußere Scheiben, zwei breite innere, durchgehender Steg).
/// Als Pfad ist es in jeder Größe scharf und nimmt die Vordergrundfarbe an.
///
/// Etwas leichter gestrichen als das Launcher-Icon, damit es bei 22 pt in der
/// Tab-Leiste nicht zuläuft.
struct FlexrDumbbell: View {

    /// Streckenzüge im 24×24-Raster: x, oben, unten, Strichstärke.
    private static let plates: [(x: CGFloat, top: CGFloat, bottom: CGFloat, width: CGFloat)] = [
        (4.2, 7.2, 16.8, 1.9),    // äußere Scheibe links
        (8.2, 4.6, 19.4, 2.6),    // innere Scheibe links
        (15.8, 4.6, 19.4, 2.6),   // innere Scheibe rechts
        (19.8, 7.2, 16.8, 1.9),   // äußere Scheibe rechts
    ]

    var body: some View {
        GeometryReader { geometry in
            let unit = min(geometry.size.width, geometry.size.height) / 24
            ZStack {
                ForEach(Self.plates.indices, id: \.self) { index in
                    let plate = Self.plates[index]
                    Path { path in
                        path.move(to: CGPoint(x: plate.x * unit, y: plate.top * unit))
                        path.addLine(to: CGPoint(x: plate.x * unit, y: plate.bottom * unit))
                    }
                    .stroke(
                        style: StrokeStyle(lineWidth: plate.width * unit, lineCap: .round, lineJoin: .round)
                    )
                }
                Path { path in
                    path.move(to: CGPoint(x: 3.4 * unit, y: 12 * unit))
                    path.addLine(to: CGPoint(x: 20.6 * unit, y: 12 * unit))
                }
                .stroke(style: StrokeStyle(lineWidth: 2 * unit, lineCap: .round))
            }
        }
        .aspectRatio(1, contentMode: .fit)
    }
}

/// Symbolsatz der App an einer Stelle gebündelt.
///
/// Wo Apple bereits ein passendes, den Nutzern vertrautes SF Symbol liefert,
/// wird es verwendet — das ist auf iOS die richtige Wahl gegenüber
/// nachgebauten Material- oder Web-Icons.
enum FlexrIcon {
    static let matches = "heart.fill"
    static let chats = "bubble.left.and.bubble.right.fill"
    static let account = "person.fill"
    static let like = "heart.fill"
    static let pass = "xmark"
    static let unmatch = "heart.slash.fill"
    static let report = "flag"
    static let block = "nosign"
    static let back = "chevron.left"
    static let forward = "chevron.right"
    static let send = "paperplane.fill"
    static let more = "ellipsis"
    static let camera = "camera.fill"
    static let locked = "lock.fill"
    static let place = "mappin.and.ellipse"
    static let close = "xmark"
    static let emoji = "face.smiling"
    static let search = "magnifyingglass"
    static let add = "plus"
    static let remove = "minus"
    static let check = "checkmark"
    static let warning = "exclamationmark.triangle.fill"
    static let eye = "eye"
    static let eyeOff = "eye.slash"
}

/// Ein Symbol in fester Kantenlänge — Hantel oder SF Symbol.
struct FlexrGlyph: View {

    enum Kind: Equatable {
        case dumbbell
        case symbol(String)
    }

    let kind: Kind
    var size: CGFloat = 20

    init(_ kind: Kind, size: CGFloat = 20) {
        self.kind = kind
        self.size = size
    }

    init(_ symbol: String, size: CGFloat = 20) {
        self.kind = .symbol(symbol)
        self.size = size
    }

    var body: some View {
        switch kind {
        case .dumbbell:
            FlexrDumbbell().frame(width: size, height: size)
        case .symbol(let name):
            Image(systemName: name)
                .font(.system(size: size * 0.86, weight: .semibold))
                .frame(width: size, height: size)
        }
    }
}
